test/test_distributions.py

```diff
@@ -1001,5 +1013,108 @@ def test_laplace_shape_tensor_params(self):
         self.assertRaises(ValueError, laplace.log_prob, self.tensor_sample_2)
 
 
+class TestNumericalStability(TestCase):
+    def _test_pdf_score(self,
+                        dist_class,
+                        x,
+                        expected_value,
+                        probs=None,
+                        logits=None,
+                        expected_gradient=None,
+                        prec=1e-5):
+        if probs is not None:
+            p = Variable(probs, requires_grad=True)
+            dist = dist_class(p)
+        else:
+            p = Variable(logits, requires_grad=True)
+            dist = dist_class(logits=p)
+        log_pdf = dist.log_prob(Variable(x))
+        log_pdf.retain_grad()
```

Comment:
You don't seem to be using `log_pdf.grad`, and even if you were it will always be all ones

Reply:
My bad, this was left over from some testing I was doing earlier. Removed.